electrumsv/gui/qt/network_dialog.py

```diff
@@ -1277,6 +1281,17 @@ def _on_menu_delete_server(self, entry: ServerListEntry) -> None:
                 self):
             return
 
+        def done_callback(future: concurrent.futures.Future[list[NetworkServerRow]]) -> None:
+            # Skip if the operation was cancelled.
+            if future.cancelled():
+                return
+            try:
+                future.result()
+            except DatabaseUpdateError as e:
+                self._main_window_weakref.show_error(e.args[0])
```

Comment:
What is this doing? Why is this the right thing to show to the user? When does it happen? What would a user need to see to make it helpful to them?

Reply:
Note that there is a `window.show_error_safe` which does the error display in a way that uses signals to pass the message to the UI thread. However, given what I said in the preceding comments I would not accept it being called here.

Reply:
Okay. I haven't actually pushed any new code yet since your first comment on this thread so I assume when you say "bad code after bad" you're referring to the hypothetical code that would be there if there was no future.result() exception check...

```
```

Reply:
> Is it an unexpected error we do not handle as things are broken in unexpected ways and the wallet can be considered broken?
^ Raise it because things are broken and hiding this level of brokenness makes it worse?